ifrs17-template/Test/TestData.ipynb

```diff
@@ -0,0 +1,366 @@
+                "\nCF,N,Calculated,4,Average,NotApplicable,EndOfPeriod,Delta,EndOfPeriod,110,1900,1",
+                "\nIA,N,Calculated,4,Average,EndOfPeriod,EndOfPeriod,Delta,EndOfPeriod,120,1900,1",
+                "\nAU,N,Optional,4,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,130,1900,1",
+                "\nEV,N,Optional,4,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,140,1900,1",
+                "\nCF,C,Optional,2,Average,NotApplicable,NotApplicable,NotApplicable,NotApplicable,160,1900,1",
+                "\nWO,C,Optional,2,Average,NotApplicable,NotApplicable,NotApplicable,NotApplicable,170,1900,1",
+                "\nCL,C,Mandatory,4,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,180,1900,1",
+                "\nEA,C,Calculated,4,EndOfPeriod,NotApplicable,NotApplicable,NotApplicable,EndOfPeriod,190,1900,1",
+                "\nAM,C,Calculated,6,EndOfPeriod,NotApplicable,NotApplicable,NotApplicable,EndOfPeriod,200,1900,1",
+                "\nFX,C,Calculated,0,NotApplicable,NotApplicable,NotApplicable,NotApplicable,NotApplicable,210,1900,1",
+                "\nEOP,C,Calculated,6,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,EndOfPeriod,220,1900,1\";"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Estimate Type"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var estimateType = ",
+                "\n@\"@@EstimateType,,,,,,,,,,,",
+                "\nSystemName,DisplayName,Order,StructureType,InputSource,PeriodType,ExternalId0,ExternalId1,ExternalId2,ExternalId3,ExternalId4,",
+                "\nBE,Best Estimate of Present Value,1,AoC,4,EndOfPeriod,,,,,,",
+                "\nAA,Advance Actuals,60,AoC,3,NotApplicable,PayablePR,ReceivableNIC,ReceivableICO,RiReceivablePR,RiPayableNIC,",
+                "\nOA,Overdue Actuals,70,AoC,3,NotApplicable,ReceivablePR,PayableNIC,PayableICO,RiPayablePR,RiReceivableNIC,",
+                "\nDA,Deferrable Actuals,80,AoC,7,NotApplicable,,,,,,",
+                "\nA,Actuals,90,None,2,NotApplicable,,,,,,",
+                "\n,,,,,,,,,,,\";"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Projection Configuration"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var projcetionConfigurationPrevious = new ProjectionConfiguration(){",
```

Comment:
why do you need projection configuration?
Are you sure these values correspond to P11? 
Do you actually need P11?

Reply:
In the TechnicalMarginTest, some scopes (ultimately TechnicalMarginForIA) call ImportStorage.GetTimeStep(0).
I double checked that values are ok (P0 is end of january).
GetTimeStep(0) calls ProjectionConfiguration[0], and ProjectionConfiguration is populated with the following query in the  ImportStorage.InitializeAsync():
(await workspace.Query<ProjectionConfiguration>()
                                               .Where(x => x.Shift > 0 || x.TimeStep == args.Month || (x.TimeStep > args.Month && x.TimeStep % periodicityInMonths == 0))
                                               .OrderBy(x => x.Shift).ThenBy(x => x.TimeStep).ToArrayAsync());
From all the configurations (0,12) seemed to be the one I wanted, as args.Month is 12 in the test case.

Reply:
In that test you don't hit that part of the code. So I guess any value would work. I suggest anyway to put all entries, not only one.

Reply:
Did in the recent commit